services/app-web/src/hooks/useScript.ts

```diff
@@ -1,22 +1,48 @@
-import { useEffect } from 'react'
+import { HTMLAttributes, useEffect } from 'react'
 
-// Add script to bottom of page, with handling for some generic boolean value to flag on or off
-const useScript = (url: string, featureFlag: boolean): void => {
+type CustomScriptParams = {
+    id: HTMLScriptElement['id'] // require ids to be able to reference specific scripts from React code
+    src: HTMLScriptElement['src'] // require src so that script file must be defined as distinct file
+    async?: HTMLScriptElement['async']
+    type?: HTMLScriptElement['type']
+    useInlineScriptNotSrc?: boolean
+} & HTMLAttributes<HTMLScriptElement>
+
+// Function to create custom scripts. Used to load third party js.
+const createScript = ({
+    type = 'text/javascript',
+    id,
+    src,
+    useInlineScriptNotSrc,
+}: CustomScriptParams): HTMLScriptElement => {
+    const script = document.createElement('script')
+    script.type = type
+    script.id = id
+    if (!useInlineScriptNotSrc) script.src = src
+    return script
+}
+
+// Add script pointing to third party URL to bottom of page, with handling for a boolean flag on or off
+const useScript = ({
+    url,
+    id,
+    featureFlag,
+}: {
+    url: string
+    id: string
+    featureFlag: boolean
```

Comment:
Is this for general purpose scripts?  Feature flags won't always be booleans.

Reply:
This is meant to be more a true false show the script or not. I will name the variable that `showFlag` instead of `featureFlag` may be clearer.  You would have to do the logic around determining the boolean in the place the hook is used e.g. if the feature flag was something that is not already true false you calculate `featureflag === THE_VALUE_THAT_MEANS_THE_FLAG` and pass that in to this parameter in hook to get a boolean value